inlineplz/interfaces/swarm.py

```diff
@@ -0,0 +1,107 @@
+# -*- coding: utf-8 -*-
+
+import random
+import subprocess
+
+import requests
+
+from inlineplz.interfaces.base import InterfaceBase
+
+class SwarmInterface(InterfaceBase):
+    def __init__(self, args):
+        """
+        SwarmInterface lets us post messages to Swarm (Helix).
+
+        args.username and args.password are the credentials used to access Swarm/Perforce.
+        args.host is the server (And any additional paths before the api)
+        args.review_id is the the review number you are commenting on
+        """
+        review_id = args.review_id
+        try:
+            review_id = int(review_id)
+        except (ValueError, TypeError):
+            print('{0} is not a valid review ID'.format(review_id))
+            return
+        self.username = args.username
+        self.password = args.password
+        self.host = args.host
+        self.topic = "review/{}".format(review_id)
+        # current implementation uses version 8 of the implementation
+        # https://www.perforce.com/perforce/doc.current/manuals/swarm/index.html#Swarm/swarm-apidoc.html#Swarm_API%3FTocPath%3DSwarm%2520API%7C_____0
+        self.version = 'v8'
+
+    def post_messages(self, messages, max_comments):
+        # randomize message order to more evenly distribute messages across different files
+        messages = list(messages)
+        random.shuffle(messages)
+
+        messages_to_post = 0
+        messages_posted = 0
+        current_comments = self.get_comments(max_comments)
+        for msg in messages:
+            if not msg.comments:
+                continue
+            messages_to_post += 1
+            body = self.format_message(msg)
+            proc = subprocess.Popen(["p4", "fstat", "-T", "depotFile", msg.path], stdout=subprocess.PIPE)
```

Comment:
i would prefer if we used `subprocess.check_output` unless theres a good reason to do it this way.